benchmarks/src/move_only_function.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <benchmark/benchmark.h>
+#include <functional>
+
+using namespace std;
+
+void mof_none(benchmark::State& state) {
+    move_only_function<void()> mof;
+    for (auto _ : state) {
+        benchmark::DoNotOptimize(mof);
+    }
+}
+
+void mof_construct(benchmark::State& state) {
+    for (auto _ : state) {
+        move_only_function<void()> mof;
+        benchmark::DoNotOptimize(mof);
+    }
+}
+
+void mof_move(benchmark::State& state) {
+    move_only_function<void()> mof;
+    for (auto _ : state) {
+        auto mof_move = move(mof);

```

Comment:
I don't understand Google Benchmark very well, but to match `mof_none`, it would seem best to `benchmark::DoNotOptimize(mof);` in the loop before the move construction.